### DISCUSSION TITLE
dependabot bump gatsby from 4.6.2 to 4.1 0.0 is breaking

### DISCUSSION THREAD

**Discussion by biantris** [gatsbyjs/gatsby#35178] [Help]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

I use @dependabot in my project (monorepo) that is in charge of updating its packages, but when trying to merge the pull request that updates gatsby from version 4.6.2 to 4.10.0 it breaks, it may be some regressive. 

We have a `test.yml` file that runs every time when dependabot opens an update pr
Running all tests within each pull request will be merged into the master branch.
It will run the yarn command to install all dependencies.
Then yarn jest will be run to run all tests.

``` shell
name: Tests

on:
  push:
    branches:
      - main
  pull_request:
    branches:
      - main

jobs:
  build:
    runs-on: ubuntu-latest

    steps:
      - uses: actions/checkout@v2
      - name: Setup Node.js
        uses: actions/setup-node@v2
        with:
          node-version: '16'
      - run: yarn
      - run: yarn jest
```

### Reproduction Link

.

### Steps to Reproduce

1. bump gatsby from 4.6.2 to 4.10.0 with @dependabot 
2. run all tests

### Expected Result

Correctly update the dependency and merge it

### Actual Result

@dependabot did not merge the update

```shell
 ● Test suite failed to run

    Cannot find module 'gatsby-page-utils/apply-trailing-slash-option' from '../../node_modules/gatsby-link/rewrite-link-path.js'
```

### Environment

```shell
"dependencies": {
    "@babel/cli": "7.17.6",
    "@babel/preset-env": "7.16.11",
    "@material-ui/core": "^4.12.3",
    "@material-ui/icons": "^4.11.2",
    "@mdx-js/mdx": "^1.6.22",
    "@mdx-js/react": "^1.6.22",
    "babel-plugin-styled-components": "^2.0.6",
    "brazilian-values": "^0.10.1",
    "formik": "^2.2.9",
    "formik-material-ui": "^1.0.0",
    "gatsby": "^4.6.2",
    "gatsby-image": "^3.11.0",
    "gatsby-plugin-canonical-urls": "^4.10.0",
    "gatsby-plugin-google-analytics": "^4.10.0",
    "gatsby-plugin-hubspot": "^2.0.0",
    "gatsby-plugin-image": "2.10.0",
    "gatsby-plugin-manifest": "^4.10.0",
    "gatsby-plugin-mdx": "^3.10.0",
    "gatsby-plugin-react-helmet": "^5.10.0",
    "gatsby-plugin-s3": "^0.3.8",
    "gatsby-plugin-sharp": "^4.10.0",
    "gatsby-plugin-sitemap": "^5.10.0",
    "gatsby-plugin-styled-components": "^5.10.0",
    "gatsby-plugin-typescript": "^4.10.0",
    "gatsby-source-filesystem": "^4.10.0",
    "gatsby-transformer-remark": "^5.10.0",
    "gatsby-transformer-sharp": "^4.10.0",
    "i18next": "^21.6.14",
    "i18next-browser-languagedetector": "^6.1.3",
    "i18next-xhr-backend": "^3.1.2",
    "isomorphic-fetch": "3.0.0",
    "prop-types": "^15.8.1",
    "qrcode.react": "2.0.0",
    "react": "17.0.2",
    "react-device-detect": "^2.1.2",
    "react-dom": "17.0.2",
    "react-helmet": "^6.1.0",
    "react-i18next": "^11.15.7",
    "react-icons": "^4.3.1",
    "react-input-mask": "^2.0.4",
    "react-qr-reader": "^2.2.1",
    "rebass": "^4.0.7",
    "styled-components": "5.3.3",
    "use-ssr": "1.0.24",
    "uuid": "8.3.2",
    "yup": "^0.32.11"
  },
```

### Config Flags

No have

**Comment by piotrkwiecinski** [comment]
Hi @biantris. It's related to changes introduced in 4.7. It's mentioned in release notes: https://www.gatsbyjs.com/docs/reference/release-notes/v4.7/#trailingslash-option 
You'll have to update jest `moduleNameMapper` configuration https://www.gatsbyjs.com/docs/how-to/testing/unit-testing/#2-creating-a-configuration-file-for-jest
